CompoundV3-Bots/Timelock-Transaction-Execution/src/agent.ts

```diff
@@ -0,0 +1,169 @@
+    const block = txEvent.blockNumber;
+
+    const creationFetchingBlockRange = networkManager.get("creationFetchingBlockRange");
+    const creationFetchingBlockStep = networkManager.get("creationFetchingBlockStep");
+    const bridgeReceiver = new ethers.Contract(
+      networkManager.get("bridgeReceiverAddress"),
+      bridgeReceiverIface,
+      provider
+    );
+    const timelock = await bridgeReceiver.localTimelock({ blockTag: block });
+
+    // Filter ExecuteTransaction logs from the local timelock and ProposalExecuted logs from the bridge receiver
+    const timelockExecutionLogs = executeTransactionLogs.filter(
+      (el) => el.address.toLowerCase() === timelock.toLowerCase()
+    );
+    const proposalExecutedLogs = txEvent.filterLog(PROPOSAL_EXECUTED_ABI, bridgeReceiver.address);
+
+    if (!timelockExecutionLogs.length && !proposalExecutedLogs.length) {
+      return [];
+    }
+
+    // Tries fetching past ProposalCreated events to get proposal calls
+    const proposalCreationInfos = (
+      await getPastEventLogs(
+        bridgeReceiver.filters.ProposalCreated(),
+        block,
+        creationFetchingBlockRange,
+        creationFetchingBlockStep,
+        provider
+      )
+    ).map((log) => bridgeReceiverIface.parseLog(log).args) as unknown as Array<ProposalCreatedArgs>;
+
+    // Processes the above data into a better format for matching, and checks whether any proposal creation
+    // event was missed (edge case)
+    const missedProposalIds: ethers.BigNumber[] = [];
+    const executedProposalInfos = proposalExecutedLogs.map((log) => {
+      const creationInfo = proposalCreationInfos.find((el) => el.id.eq(log.args.id));
+
+      if (!creationInfo) {
+        missedProposalIds.push(log.args.id);
+        return undefined;
+      }
+
+      return {
+        id: creationInfo.id,
+        calls: creationInfo.targets.map((_, idx) => ({
+          matched: false,
+          target: creationInfo.targets[idx],
+          value: creationInfo.txValues[idx],
+          signature: creationInfo.signatures[idx],
+          data: creationInfo.calldatas[idx],
+        })),
+        eta: creationInfo.eta,
+      };
+    });
+
+    if (missedProposalIds.length) {
+      console.warn(
+        "Current creationFetchingBlockRange parameter is too low, so some proposal creations couldnt'be fetched. Please consider increasing it."
+      );
+      console.warn(`Missed proposal IDs: ${missedProposalIds.map((id) => id.toString()).join(", ")}`);
+    }
+
+    const timelockExecutionInfos = timelockExecutionLogs.map((log) => ({
+      matched: false,
+      ...log.args,
+    })) as unknown as Array<ExecuteTransactionArgs & { matched: boolean }>;
+
+    // Matches proposal calls with executed calls
+    timelockExecutionInfos.forEach((callExecution) => {
+      for (const proposalExecution of executedProposalInfos) {
+        if (proposalExecution && !proposalExecution.eta.eq(callExecution.eta)) {
+          continue;
+        }
```

Comment:
this condition can be updated to skip the loop if `proposalExecution` is undefined: 
` if (!proposalExecution || !proposalExecution.eta.eq(callExecution.eta))`

Reply:
Good call!

Reply:
Done in [d03da1d](https://github.com/NethermindEth/Forta-Agents/pull/390/commits/d03da1d0290e87d7397992cbf5bc1e0a314b6550).